projects/angular/src/forms/datepicker/providers/date-navigation.service.ts

```diff
@@ -68,9 +83,8 @@ export class DateNavigationService {
    * Notifies that the selected day has changed so that the date can be emitted to the user.
    * Note: Only to be called from day.ts
    */
-  notifySelectedDayChanged(dayModel: DayModel) {
-    this.selectedDay = dayModel;
-    this._selectedDayChange.next(dayModel);
+  notifySelectedDayChanged(dayModel: DayModel, emitEvent: boolean) {
+    this.compareAndSelectDate(dayModel, emitEvent);
```

Comment:
```suggestion
    if (dayModel !== null) {
      if (emitEvent) {
        this.setSelectedDay(dayModel);
      } else {
        this.setInterimSelectedDay(dayModel);
      }
    }
```

The `compareAndSelectDate` method is only called here, and that method is the only place that the `updateSelectedDay` method is called. I would inline here (and remove the `startDate` stuff because it's not needed here).